common/interactivetx.c

```diff
@@ -0,0 +1,687 @@
+#include "config.h"
+#include "interactivetx.h"
+#include <bitcoin/chainparams.h>
+#include <bitcoin/psbt.h>
+#include <bitcoin/script.h>
+#include <ccan/array_size/array_size.h>
+#include <ccan/cast/cast.h>
+#include <ccan/mem/mem.h>
+#include <ccan/tal/str/str.h>
+#include <common/billboard.h>
+#include <common/blockheight_states.h>
+#include <common/channel_type.h>
+#include <common/gossip_rcvd_filter.h>
+#include <common/gossip_store.h>
+#include <common/initial_channel.h>
+#include <common/lease_rates.h>
+#include <common/memleak.h>
+#include <common/peer_billboard.h>
+#include <common/peer_failed.h>
+#include <common/peer_io.h>
+#include <common/psbt_internal.h>
+#include <common/psbt_open.h>
+#include <common/read_peer_msg.h>
+#include <common/setup.h>
+#include <common/status.h>
+#include <common/subdaemon.h>
+#include <common/type_to_string.h>
+#include <common/wire_error.h>
+
+#define NUM_TX_MSGS (TX_RM_OUTPUT + 1)
+enum tx_msgs {
+	TX_ADD_INPUT,
+	TX_ADD_OUTPUT,
+	TX_RM_INPUT,
+	TX_RM_OUTPUT,
+};
+
+/*
+ * BOLT-f53ca2301232db780843e894f55d95d512f297f9 #2:
+ * The maximum inputs and outputs are capped at 252. This effectively fixes
+ * the byte size of the input and output counts on the transaction to one (1).
```

Comment:
i don't think we actually check for this in this file? maybe we should, in the tx builder after tx_complete is called?

Reply:
Oh hm that's a great question

Reply:
good idea 👍